File: haskell/greet/greet.hs
-- File: Greet.hs
module Main where

import System.Environment (getArgs)

main :: IO ()
main = do
    args <- getArgs
    case args of
        [name] -> putStrLn $ "Hello, " ++ name ++ "!"
        _      -> putStrLn "Usage: greet <name>"

File: elixir/greet/greet.exs
# greet.exs
defModule Greeter do
  def greet(name) do
    IO.puts("Hi, #{name}!")
  end
end

defmodule CLI do
  def main(args) do
    case args do
      [name] ->
        Greeter.greet(name)
      _ ->
        IO.puts("Usage: elixir greet.exs <name>")
      end
    end
end

CLI.main(System.argv())
